linux/net/rina/dtcp.c

```diff
@@ -747,9 +1125,32 @@ static int rcv_flow_ctl(struct dtcp * dtcp,
         pci = pdu_pci_get_rw(pdu);
         ASSERT(pci);
 
-        snd_rt_wind_edge_set(dtcp, pci_control_new_rt_wind_edge(pci));
+        rt = pci_control_sndr_rate(pci);
+        tf = pci_control_time_frame(pci);
+
+        // Is this a window based control flow?
+        if(dtcp_window_based_fctrl(dtcp_config_get(dtcp))) {
+        	snd_rt_wind_edge_set(dtcp, pci_control_new_rt_wind_edge(pci));
+        	dt_sv_window_closed_set(dtcp->parent, false);
```

Comment:
We need to distinguish when using window- or rate-based FC, and then react to the receiving of the flow control in a separate way. Makes no sense to update the right window edge when we have rate-based FC(which does not care of windows).

If the flow is not aligned here then we can have, in the same dif, two IPCP with a flow emitting at different speed (depending on which node you are looking at). The flow is unique and the endpoints of such flow are maintained aligned on the same rate.